pandas/io/formats/format.py

```diff
@@ -1912,22 +1914,20 @@ def _trim_zeros_complex(str_complexes: np.ndarray, decimal: str = ".") -> list[s
     Separates the real and imaginary parts from the complex number, and
     executes the _trim_zeros_float method on each of those.
     """
-    trimmed = [
-        "".join(_trim_zeros_float(re.split(r"([j+-])", x), decimal))
-        for x in str_complexes
-    ]
+    real_part, imag_part = [], []
+    for x in str_complexes:
+        trimmed = re.split(r"([j+-])", x)
+        real_part.append("".join(trimmed[:-4]))
```

Comment:
Could you add a comment here why you are using a constant slice of `:-4` and `-4:-2`?